### PR TITLE
Remodel address form from checkout, add validation for the state field

### PR DESCRIPTION
**Pull request by AlexeyKaryka**
Update Input component, address form from the checkout, add validation for the state field

## This PR is a:

[ ] New feature
[ ] Enhancement/Optimization
[ X ] Refactor
[ ] Bugfix
[ ] Test for existing code
[ ] Documentation

Add validation block to the address form and show details about invalid address form fulfilling there.

## Summary

When this pull request is merged, it will fix issue with saving address in checkout https://github.com/magento-research/pwa-studio/issues/462 and Common Input does not handle initial value correctly https://github.com/magento-research/pwa-studio/issues/416

### CODE REVIEW

**Comment by zetlen** [comment]
Hey @AlexeyKaryka, could you please link to the issue this resolves? You can leave a comment in a commit message like "Fixes #NNN" with the issue number. Then, we can easily open the deployed app and do UAT.

**Comment by Starotitorov** [comment]
@zetlen here is a link to the issue https://github.com/magento-research/pwa-studio/issues/462, sorry

**Comment by zetlen** [comment]
Oh excellent, thank you!

**Comment by Starotitorov** [comment]
I guess it will be better to use asField HOC provided by informed and wrap our input component in this HOC instead of passing such properties as fieldValue, setFieldValue to common Input component in the implementation of getInput method.

**Comment by Starotitorov** [comment]
@AlexeyKaryka In general, it is really great that you removed `value` and `dirty` properties from local state of our Input component because informed provide us this information and there is no sense to duplicate it.

File: packages/venia-concept/src/components/Checkout/address.js [hunk 1]
```diff
@@ -1,52 +1,104 @@
...
+        const { classes } = this.props;
+        const inputClasses = {
+            input: classes.textInput,
+            root: classes.inputRoot,
+            rootFocused: classes.inputRootFocused
+        };
+        const validator = this.getValidator(fieldName);
+
+        return (
+            <div className={classes[fieldKey]}>
```
Comment (pcvonz) [inline packages/venia-concept/src/components/Checkout/address.js]: Need to add a `key` to this div.

File: packages/venia-concept/src/components/Checkout/flow.css [hunk 2]
```diff
@@ -10,6 +10,7 @@
     bottom: 5.5rem;
     box-shadow: 0 -1px rgb(var(--venia-border));
     display: grid;
+    grid-template-columns: 50% 50%;
```
Comment (pcvonz) [inline packages/venia-concept/src/components/Checkout/flow.css]: Don't think we want the checkout options to look like this:
image

File: packages/venia-concept/src/components/CreateAccount/createAccount.js [hunk 3]
```diff
@@ -43,25 +47,28 @@ class CreateAccount extends Component {
...
 
     get isIncompleteOrInvalid() {
         // A catch-all to check if the form is valid so the customer can attempt to create an account
         return (
             this.hasEmailError ||
             this.hasPasswordConfirmError ||
-            !this.state.email ||
-            !this.state.firstName ||
-            !this.state.lastName
+            !this.getFieldValue(fields.email) ||
```
Comment (pcvonz) [inline packages/venia-concept/src/components/CreateAccount/createAccount.js]: It'd be great if all of this was checked through `Informed`. Separate issue probably.

File: packages/venia-concept/src/components/CreateAccount/createAccount.js [hunk 4]
```diff
@@ -174,14 +192,15 @@ class CreateAccount extends Component {
     }
 
     onCreateAccount = () => {
+        const { getFieldValue } = this;
```
Comment (pcvonz) [inline packages/venia-concept/src/components/CreateAccount/createAccount.js]: You can just pass `formState` into this function. I'm not sure why all of these fields are kept in `state`.

File: packages/venia-concept/src/components/Input/input.css [hunk 5]
```diff
@@ -25,29 +17,61 @@
...
+}
+
 .root {
     padding: 0.5rem 1.5rem;
+    --input-right-padding: 1rem;
+}
+
+.rootFocused {
+    composes: root;
+    background-color: rgb(var(--venia-teal-alt));
```
Comment (pcvonz) [inline packages/venia-concept/src/components/Input/input.css]: Follow: https://github.com/magento-research/pwa-studio/issues/517. Should have a `background-color`, etc.

File: packages/venia-concept/src/components/Input/input.js [hunk 6]
```diff
@@ -82,52 +111,83 @@ class Input extends Component {
...
-                    title={title}
-                    autoComplete={autoComplete}
-                    onChange={this.handleChange}
-                    onFocus={this.focusTextInput}
-                    onBlur={this.blurTextInput}
-                    field={field}
-                />
-                {helpText}
+                <div className={classes.inputContainer}>
+                    <BasicText
```
Comment (pcvonz) [inline packages/venia-concept/src/components/Input/input.js]: I think we should have the option to pass in any `Informed` component, just wrapping `BasicText` really limits what we can do with `Informed`. This `Input` component could be made a lot more generic. 

For example, I think that the State field should use the `Select` component from `Informed` instead. Removes the need to have that error display too.

File: packages/venia-concept/src/components/CreateAccount/createAccount.js (hunk 4 above)
Comment (pcvonz) [inline packages/venia-concept/src/components/CreateAccount/createAccount.js]: You can just use `formState` much how you're passing `formState` into `handleFormState` here. I'm not sure why the whole form is duplicated into the component `state`.

File: packages/venia-concept/src/components/Checkout/flow.css (hunk 2 above)
Comment (pcvonz) [inline packages/venia-concept/src/components/Checkout/flow.css]: Why was this added? It changed the layout of each section of checkout.

File: packages/venia-concept/src/components/Input/input.js [hunk 7]
```diff
@@ -82,52 +111,83 @@ class Input extends Component {
...
+                onClick={this.resetValue}
+            >
+                <Icon name={name} attrs={attrs} />
+            </button>
+        );
+    }
+
+    resetValue = () => {
+        const { fieldApi } = this.props;
+        fieldApi.setValue(null);
```
Comment (Starotitorov) [inline packages/venia-concept/src/components/Input/input.js]: You need to call onChange here.
